File: Pos.Api/Entities/Printer.cs
namespace Pos.Api.Entities;

public class Printer
{
    public Guid Id { get; set; }
    public Guid RestaurantId { get; set; }
    public Guid StoreId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string IpAddress { get; set; } = string.Empty;
    public int Port { get; set; }
    public string PrinterType { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }

    public Restaurant Restaurant { get; set; } = null!;
    public Store Store { get; set; } = null!;
}

File: Pos.Till.App/lib/models/printer.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'printer.freezed.dart';
part 'printer.g.dart';

/// Printer type string the API stores: "Receipt" / "Kitchen".
class PrinterType {
  static const String receipt = 'Receipt';
  static const String kitchen = 'Kitchen';
}

@freezed
class Printer with _$Printer {
  const factory Printer({
    required String id,
    required String storeId,
    required String name,
    required String ipAddress,
    required int port,
    required String printerType,
    required bool isActive,
    required DateTime createdAt,
  }) = _Printer;

  factory Printer.fromJson(Map<String, dynamic> json) =>
      _$PrinterFromJson(json);
}
